src/task.c

```diff
@@ -116,6 +116,7 @@ fetch_finish_callback (GError *error, guint32 match_cnt,
         } else {
             g_propagate_error (&task->error, error);
             task->state = TASK_COMPLETE;
+            task->fetch_succeeded = false;
```

Comment:
What if you did,
          app_data->aborted = ABORTED_TASK;
          g_cancellable_cancel(app_data->cancellable);
THEN  (see comments further)

Reply:
Hi @cbouchar And thanks a lot for comment.

Your are right this solution would be better and in fact it was the first solution I have tried. But it turned out not to be working. In this solution the task is corectly aborted with the correct abort message. 
I've reimplemented this solution in this commit (https://github.com/restraint-harness/restraint/pull/295/commits/325833750a4a2b51ebd789b4ca8cfb6069535a73) and tried to run it beaker. And even though the task is aborted correctly the whole recipe just continue to run and the following tasks are executed as normal. 
I haven't been able to solve this issue in the past so I've come up with the solution you've been reviewing.

So the actual problem with this solution is that the recipe is not aborted and conitues to run even though the RESTRAINT_ERROR is properly set. 
I'll look at this aborting issue once again and try to solve it, but If you have any idea why the recipe is not aborted I'd like to hear your insights.

Thanks!

Reply:
Not sure you saw my 2nd proposal.  This sounds like an extreme corner case for a specific area in your recipe.  It seems more reasonable to create a task to check if your requested test got installed. if not just call rstrnt-abort? Messing with restraint will affect all users and you'd be surprised the bugs that can surface.

Reply:
Yes, I saw your second proposal. If there won't be any possibility to push this update upstream, we would need to implement some kind of wrapper. However, even this wrapper could fail to fetch data, which would result in the same outcome as if there were no wrapper at all.

Please take a quick look at my latest changes. I've refactored the code to minimize disruption and make the new feature as optional as possible for the end user.

Regarding this feature, you may see it as an extreme corner case. But is it? All we want to do is abort the recipe if there is a fetch error in some task. As it stands, the user currently has no way to do that in restraint. This provides the user with an option—not a necessity—to abort the recipe, which could result in saving computation time and stopping tasks that would produce irrelevant data.

And yes, I know there's always the possibility that some bugs could surface, but as I've said, this feature is designed to be as optional as possible.

Thanks a lot for your time!

Reply:
My suggestion wasn't to retry a fetch but to check if the directory is present.  if not, abort.  I'm concerned about disrupting other engineers.  BTW I'll be on PTO next week.

Reply:
After discussion, let postpone this pull request for Q1. Thanks!